core/src/main/java/com/linecorp/armeria/server/RoutingTrie.java

```diff
@@ -62,42 +66,73 @@
                                                                null, null, ImmutableList.of());
 
     private final Node<V> root;
+    private final Function<V, Route> routeResolver;
 
     @SuppressWarnings("unchecked")
     private Node<V> continueWalking() {
         return (Node<V>) CONTINUE_WALKING;
     }
 
-    RoutingTrie(Node<V> root) {
-        requireNonNull(root, "root");
-        this.root = root;
+    RoutingTrie(Node<V> root, Function<V, Route> routeResolver) {
+        this.root = requireNonNull(root, "root");
+        this.routeResolver = requireNonNull(routeResolver, "routeResolver");
     }
 
+    // FIXME(hyangtack) Update Javadoc
+
     /**
      * Returns the list of values which is mapped to the given {@code path}.
      */
-    List<V> find(String path) {
-        final Node<V> node = findNode(path, false);
+    List<V> find(RoutingContext routingCtx) {
+        final Node<V> node = findNode(routingCtx, false, null);
         return node == null ? ImmutableList.of() : node.values;
     }
 
+    /**
+     * Returns the list of {@link Routed}s which are mapped to the given {@link RoutingContext}.
+     */
+    List<Routed<V>> findRoutes(RoutingContext routingCtx) {
+        final ImmutableList.Builder<Routed<V>> routeAccumulator = new Builder<>();
+        findNode(routingCtx, false, routeAccumulator);
+        return routeAccumulator.build();
+    }
+
     /**
      * Returns the list of values which is mapped to the given {@code path}.
      */
-    List<V> findAll(String path) {
-        return findAllNodes(path, false)
+    List<V> findAll(RoutingContext routingCtx) {
+        return findAllNodes(routingCtx, false, null)
                 .stream()
                 .flatMap(n -> n.values.stream())
                 .collect(toImmutableList());
     }
 
+    /**
+     * Returns the list of {@link Routed}s which are mapped to the given {@link RoutingContext}.
+     */
+    List<Routed<V>> findAllRoutes(RoutingContext routingCtx) {
+        final ImmutableList.Builder<Routed<V>> routeAccumulator = new Builder<>();
+        findAllNodes(routingCtx, false, routeAccumulator);
+        return routeAccumulator.build();
+    }
```

Comment:
Is it okay to replace the existing `find*` methods with these newly added ones?
Please let me know what you think/concern. :-)